stake-pool/js/rollup.config.js

```diff
@@ -0,0 +1,157 @@
+import babel from '@rollup/plugin-babel';
+import commonjs from '@rollup/plugin-commonjs';
+import json from '@rollup/plugin-json';
+import nodeResolve from '@rollup/plugin-node-resolve';
+import replace from '@rollup/plugin-replace';
+import {terser} from 'rollup-plugin-terser';
+
+const env = process.env.NODE_ENV;
+const extensions = ['.js', '.ts'];
+
+function generateConfig(configType, format) {
+  const browser = configType === 'browser';
+  const bundle = format === 'iife';
+
+  const config = {
+    input: 'src/index.ts',
+    plugins: [
+      commonjs(),
+      nodeResolve({
+        browser,
+        dedupe: ['bn.js', 'buffer'],
+        extensions,
+        preferBuiltins: !browser,
+      }),
+      babel({
+        exclude: '**/node_modules/**',
+        extensions,
+        babelHelpers: bundle ? 'bundled' : 'runtime',
+        plugins: bundle ? [] : ['@babel/plugin-transform-runtime'],
+      }),
+      replace({
+        preventAssignment: true,
+        values: {
+          'process.env.NODE_ENV': JSON.stringify(env),
+          'process.env.BROWSER': JSON.stringify(browser),
+        },
+      }),
+    ],
+    onwarn: function (warning, rollupWarn) {
+      if (warning.code !== 'CIRCULAR_DEPENDENCY') {
+        rollupWarn(warning);
+      }
+    },
+    treeshake: {
+      moduleSideEffects: false,
+    },
+  };
+
+  if (configType !== 'browser') {
+    // Prevent dependencies from being bundled
+    config.external = [
+      /@babel\/runtime/,
+      '@solana/buffer-layout',
+      'bn.js',
+      'borsh',
+      'bs58',
+      'buffer',
+      'crypto-hash',
+      'jayson/lib/client/browser',
+      'js-sha3',
+      'cross-fetch',
+      'rpc-websockets',
+      'secp256k1',
+      'superstruct',
+      'tweetnacl',
+    ];
```

Comment:
I think these and the ones just below should be the dependencies that are explicitly declared in `package.json`, ie:
```
    "@project-serum/borsh": "^0.2.2",
    "@solana/buffer-layout": "^4.0.0",
    "@solana/spl-token": "^0.1.8",
    "@solana/web3.js": "^1.30.2",
    "bn.js": "^5.2.0",
    "buffer": "^6.0.3"
```